### ISSUE TITLE
Visual Studio 2019 creates invalid assembly name

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/aspnetcore#26343]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1165370/visual-studio-2019-creates-invalid-assembly-name.html)._

---
[severity:I'm somewhat annoyed. A fix would be nice]
I tried to create a ASP.NET Core 3.1 Web App using the following project/solution name: "Can'tDebugFF".

Visual Studio apparently doesn't escape the project's assembly name.

The linker, too, doesn't complain about the illegal assembly name.

Debugging the web app then results in:

```
FileLoadException: The given assembly name or codebase was invalid. (0x80131047)

System.Reflection.AssemblyName.nInit()
```

Please fix by correctly escaping assembly name on project creation.

---
#### Feedback Bot on 8/27/2020, 08:13 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

---
### Original Solutions
(no solutions)

**Comment by Pilchie** [comment]
@dsplaisted thoughts on how this might work?  It looks like the template engine embeds an escaped version of `<RootNamespace>` in the project file here, but since the assembly name comes from the project file it doesn't work.  Could the template engine also put something to escape the assembly name?

**Comment by dsplaisted** [comment]
Yes, we could consider that.  I'm guessing we don't do it now because most commonly it would be a space that needs to be escaped for the namespace, and maybe that's not necessary for the assembly name.

We could also do it automatically in the SDK, but cleaning up or escaping names in MSBuild logic isn't great to write.  See dotnet/sdk#12562 where we handle spaces in the root namespace.

**Comment by Pilchie** [comment]
Also tagging @vijayrkn - do you know where the escaping of `RootNamespace` happens?

**Comment by Pilchie** [comment]
Note - as a workaround, it's already possible to specify `<AssemblyName>Can_tDebugFF</AssemblyName>` in the project file.
